### ISSUE TITLE
Suggestion: it does not support PC joysticks such as TGZ, BETOP, BGM gamepad

### ISSUE THREAD

**Issue by xuancong84** [Ryochan7/DS4Windows#2933]
What a wonderful program for both switching among controllers and binding button to keyboard.
However, when I plug in my PC gamepad that can switch mode to emulate TGZ, BETOP, and BGM gamepad respectively, DS4Windows cannot detect either of them.
Nevertheless, the gamepad tester website (https://gamepad-tester.com/) can detect all of TGZ, BETOP, and BGM.

**Comment by Ryusennin** [comment]
DS4Windows doesn't support generic PC controllers.
It is a program to make your DS4/DS5 (or compatible device) appear as a native XInput controller.

**Comment by xuancong84** [comment]
The problem I encountered is that I have PC joysticks (that can switch mode between TGZ, BFM, Betop), PS4 joysticks, Nintendo switch joysticks, etc. Even without DS4Windows or any other programs, they can connect to Windows 10. However, in the same game, the button assignment is different. I want to emulate the same joystick no matter what physical joystick I plugged in, so that the button assignment will be the same. Otherwise, every time I plugged in a different joystick, I need to reconfigure the button settings in every game so as to comform the same button assignment. Is there any other program I can use for this purpose?
